iFixit/react-commerce
# Hide empty product list children (server-side version)

Hides empty product list children on parts product list pages.

Makes an extra request to Algolia on the server only to identify all child devices with products. This is used to filter the child devices collected from Strapi.

## QA

Examples of collections with no products:

https://www.ifixit.com/Parts/Computer_Hardware
https://www.ifixit.com/Parts/Samsung_Galaxy_S22

Also make sure that no collections that are non-empty end up being hidden by this change.

## References

https://www.algolia.com/doc/api-reference/api-methods/search/
https://www.algolia.com/doc/api-reference/api-parameters/filters/
https://support.algolia.com/hc/en-us/articles/5588550618897-Clarifying-confusion-with-facets-facet-values-maxFacetHits-and-maxValuesPerFacet

CC @dhmacs @sterlinghirsh 

Closes #822
